src/scales/scale.ts

```diff
@@ -122,19 +118,23 @@ module Plottable {
       }
     }
 
-    protected _getDomain() {
-      return this._d3Scale.domain();
+    protected _getDomain(): D[] {
+      throw new Error("subclasses should override _getDomain");
     }
 
     protected _setDomain(values: D[]) {
       if (!this._domainModificationInProgress) {
         this._domainModificationInProgress = true;
-        this._d3Scale.domain(values);
+        this._setBackingDomain(values);
         this._dispatchUpdate();
         this._domainModificationInProgress = false;
       }
     }
 
+    protected _setBackingDomain(values: D[]) {
```

Comment:
All of these should be documented in the superclass (here). As a developer I am not even sure what is the contract I am getting into if I decide to extend the classes and I have to implement these methods.